controllers/nutanixmachine_controller.go

```diff
@@ -216,13 +257,13 @@ func (r *NutanixMachineReconciler) reconcileDelete(rctx *nctx.MachineContext) (r
 	vmName := rctx.NutanixMachine.Name
 	klog.Infof("%s Handling NutanixMachine deletion of VM: %s", rctx.LogPrefix, vmName)
 	conditions.MarkFalse(rctx.NutanixMachine, infrav1.VMProvisionedCondition, capiv1.DeletingReason, capiv1.ConditionSeverityInfo, "")
+	vmUUID := getVMUUID(ctx, rctx.NutanixMachine)
 	// Check if VMUUID is absent
-	if rctx.NutanixMachine.Status.VmUUID == "" {
+	if vmUUID == "" {
```

Comment:
Have an early return for this case and remove the else case outside the conditional clause

Reply:
Let's keep these changes for a different PR